js/.yarn/sdks/prettier/index.js

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env node
+
+const {existsSync} = require(`fs`);
+const {createRequire, createRequireFromPath} = require(`module`);
+const {resolve} = require(`path`);
+
+const relPnpApiPath = "../../../.pnp.cjs";
+
+const absPnpApiPath = resolve(__dirname, relPnpApiPath);
+const absRequire = (createRequire || createRequireFromPath)(absPnpApiPath);
+
+if (existsSync(absPnpApiPath)) {
+  if (!process.versions.pnp) {
+    // Setup the environment to be able to require prettier/index.js
+    require(absPnpApiPath).setup();
+  }
+}
+
+// Defer to the real prettier/index.js your application uses
+module.exports = absRequire(`prettier/index.js`);
```

Comment:
There should be a message stating it's out of date. The reasoning behind it is to prevent inconsistencies where people use different versions of package managers, similar to inconsistencies we had when NPM 7 released and they introduced a new lockfile format.

Reply:
Yes, but it's only needed for Prettier and TS integration with VSCode and other IDEs.

Imo I think this is worth committing because it creates a working out of the box dev environment.

The code can be updated automatically vwith `yarn dlx @yarnpkg/sdks`, though the VS Code config file needs to be altered slightly in order to play nicely with our js folder being a subdirectory.

https://yarnpkg.com/getting-started/editor-sdks

Reply:
My concern is keeping this up to date across our (many) repos. What manual changes are needed, and could we automate them somehow?

Reply:
> By default, it generates a vscode config file in the JS folder. We just move that out, and prepend `js/` to any paths needed.
> 
> Looking now, I can't actually see the vscode config file in this PR... I need to add that.

I suppose we could add something to flarum-cli. But I wonder if the huge pile of boilerplate we need to keep track of is worth it?

Reply:
> It's one of the things that won't be a 'huge pile' when we go monorepo eventually. Our root workspace could have Prettier installed with this boilerplate, and everything would work fine in subdirectories.

It's not just us: monorepo or not, we can't sacrifice developer experience for third party devs.

Reply:
Just a note that these changes you're referring to are specifically for Yarn PnP support rather than Yarn 2 or Yarn 3.

Reverting 9781c8c will let us use Yarn 3 without PnP using the legacy node_modules linker.

Reply:
To be honest, if we want to commit entirely to Yarn, I would prefer switching to Yarn 3 without PnP, and then reconsider PnP at a later time. Incremental steps!